tasks/modules/logging/collins.rb

```diff
@@ -10,7 +10,10 @@ def self.facter
     end
 
     def self.log message, level = nil
-      if facter['asset_tag']
+      asset_tag = facter['asset_tag']
+
+      # The AssetCreation task may try to log before the asset exists in Collins
+      if asset_tag && collins.exists?(asset_tag)
```

Comment:
this will double the number of RPCs to collins to log. During induction, loads of log messages are generated, so this will create a vast uptick in load on Collins. Perhaps caching the existence of the asset in collins makes more sense?